File: example/kotlinlib/script/3-custom-class/Qux.kt
//| extends: [millbuild.LineCountKotlinModule]
//| kotlinVersion: 2.0.20
package qux

fun getLineCount(): String =
    ::main.javaClass.classLoader
        .getResourceAsStream("line-count.txt")
        .readAllBytes()
        .toString(Charsets.UTF_8)

fun main() {
    println("Line Count: " + getLineCount())
}

File: example/scalalib/script/7-custom-class/Qux.scala
//| extends: [millbuild.LineCountScalaModule]
//| scalaVersion: 3.7.3
package qux

def getLineCount() = {
  scala.io.Source
    .fromResource("line-count.txt")
    .mkString
}

@main def main() = {
  println(s"Line Count: ${getLineCount()}")
}
